cmd/lakectl/cmd/local_checkout.go

```diff
@@ -54,6 +54,12 @@ func localCheckout(cmd *cobra.Command, localPath string, specifiedRef string, co
 		DieErr(err)
 	}
 
+	force := Must(cmd.Flags().GetBool(localForceFlagName))
+	dieOnInterruptedOperation(idx.Operation, force)
+	_, err = local.WriteOperation(localPath, "checkout")
```

Comment:
I suggest instead of a WriteOperation/RemoveOperationFromIndexFile functions simply pass the operation when performing WriteIndex (which can be empty).